dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/file/batch/BatchFileOrchestrator.kt

```diff
@@ -56,6 +56,17 @@ internal class BatchFileOrchestrator(
         return reusableFile ?: createNewFile()
     }
 
+    override fun getNewWritableFile(): File? {
```

Comment:
instead of making a new method that is almost identical to `getWritableFile`  you could instead do 

```
    @WorkerThread
    override fun getWritableFile(forceNewFile: Boolean = false): File? {
        if (!isRootDirValid()) {
            return null
        }
        deleteObsoleteFiles()
        freeSpaceIfNeeded()
        return if (!forceNewFile){
            getReusableWritableFile() ?: createNewFile()
        } else {
            createNewFile()
        }
    }
```

Reply:
Yes, was not very sure which way to go here. Thanks for sharing your opinion, will change that :)